test/test_response.py

```diff
@@ -125,6 +126,45 @@ def test_get_all_memory_usage_single_chunk(self) -> None:
         buffer.put(chunk)
         assert buffer.get_all() is chunk
 
+    @pytest.mark.parametrize(
+        "get_func",
+        (lambda b: b.get(1024), lambda b: b.get_all()),
+        ids=("get", "get_all"),
+    )
+    def test_performance_multi_chunk(
+        self, get_func: typing.Callable[[BytesQueueBuffer], str]
+    ) -> None:
+        buffer = BytesQueueBuffer()
+        now = time.perf_counter()
+
+        for _ in range(10 * 1024):  # 10 MiB total
+            buffer.put(b"x" * 1024)  # 1 KiB
+        while len(buffer) > 0:
+            get_func(buffer)
+
+        delta = time.perf_counter() - now
+
+        assert delta < 0.5, f"Data put & get time too long: {delta}s"
+
+    @pytest.mark.parametrize(
+        "get_func",
+        (lambda b: b.get(1024), lambda b: b.get_all()),
+        ids=("get", "get_all"),
+    )
+    def test_performance_single_chunk(
+        self, get_func: typing.Callable[[BytesQueueBuffer], str]
+    ) -> None:
+        buffer = BytesQueueBuffer()
+
+        now = time.perf_counter()
+        buffer.put(b"x" * 1024 * 1024 * 10)  # 10 MiB
+        while len(buffer) > 0:
+            get_func(buffer)
+
+        delta = time.perf_counter() - now
+
+        assert delta < 0.5, f"Data put & get time too long: {delta}s"
```

Comment:
The tests that assert on the time delta might be a bit fragile across different environments. Machines with different performance characteristics can cause these timing-based tests to pass or fail inconsistently.

The patched version does clearly improve performance, though: on my machine `TestBytesQueueBuffer.test_performance_single_chunk[get]` shows that it allocates almost half as much memory. Would it make sense to rely on this and add an explicit test that checks the memory allocation instead of the timing-based tests?